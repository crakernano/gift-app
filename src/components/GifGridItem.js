import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = (img)=>{
    return(
        <div className="card animate_fadeIn">
            <img src={img.url} alt={img.title} />
            <p>{img.title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}