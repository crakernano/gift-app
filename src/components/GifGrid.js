import React from 'react'
import { useFeatchGifs } from '../hooks/useFeatchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

//    const [images, setImages] = useState([]);


    const {data, loading} = useFeatchGifs( category );

    return (
        <div className="card-grid">
            <h3>{category}</h3>
            {loading ? 'Cargando...' : ''}
                {
                data.map(img => (
                    <GifGridItem 
                        key={img.id}
                        { ...img }
                    />
                ))
                }
        </div>
    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}