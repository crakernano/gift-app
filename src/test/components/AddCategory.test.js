import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory'

describe('Pruebas en <AddCategory />', ()=>{

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategorias={setCategories}/>);

    beforeEach(()=>{
        jest.setCategorias = jest.fn();
        wrapper = shallow(<AddCategory setCategorias={setCategories}/>);
    })
    test('Debe mostrarse correctamente', ()=>{

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe cambiar la caja de texto', ()=>{

        const input = wrapper.find('input');
        const value = 'value'
        input.simulate('change', {target:{value}});

        expect(wrapper).toMatchSnapshot();
    })

    test('No debe mandar la informaciÃ³n', ()=>{
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    })

    test('Debe llamar al setCategories y vaciar la caja de texto', ()=>{
        const value = 'Hola';

        wrapper.find('input').simulate('change', {target:{value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    })

})