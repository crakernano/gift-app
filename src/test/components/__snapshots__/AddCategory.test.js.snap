// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe cambiar la caja de texto 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="value"
  />
</form>
`;

exports[`Pruebas en <AddCategory /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
